_data/sources/meetup.js

```diff
@@ -1,17 +1,63 @@
-const fetch = require("node-fetch"),
+const {GraphQLClient, gql} = require("graphql-request"),
     group = require("./templates/group"),
     event = require("./templates/event");
 
 class meetup {
-    constructor() {
+    constructor(token = process.env.MEETUP_TOKEN) {
         this.groups = require("./groupIds/meetup").filter((value, index, self) => self.indexOf(value) === index);
-        this.apiEvents = (group) => 'https://api.meetup.com/' + group + '/events';
+        this.graphQLClient = new GraphQLClient('https://api.meetup.com/gql', {
+            headers: {
+                authorization: 'Bearer ' + token,
+                method: 'POST',
+                dataType: 'jsonp'
+            },
+        });
+        this.apiEvents = (group) => gql`
+            {
+                groupByUrlname(urlname: ${group}) {
+                    unifiedEvents {
+                        edges {
+                            node {
+                                title
+                                shortUrl
+                                venue
+                                description
+                                dateTime
+                                endTime
+                                going
+                                maxTickets
+                                fees
+                                price
+                                groupName
+                                groupLink
+                                source
+                                ad
```

Comment:
Same with the above comment, this does not come from the API, is by default `false` and is a placeholder encase any individual or organisation wants to donate to the project and have their events / group promoted (like an ad, as it were, which can be highlighted differently on the front-end).

Reply:
resolved by e57dccf